### DISCUSSION TITLE
How to start MainWindowViewModel properly?

### DISCUSSION THREAD

**Discussion by GoldenWatch** [AvaloniaUI/Avalonia#6368] [Q&A] [answered]
The final step in https://docs.avaloniaui.net/guides/deep-dives/reactiveui/routing, BuildAvaloniaApp().Start<MainWindow>(() => new MainWindowViewModel()); is deprecated and a different suggestion is made in https://github.com/avaloniaui/Avalonia/wiki/Application-lifetimes, what's the current correct way to start the app with MainWindowViewModel?

**Comment by jp2masa** [comment, marked as answer]
Something like https://github.com/AvaloniaUI/Avalonia/blob/master/samples/Sandbox/App.axaml.cs#L14-L20 with:

```cs
new MainWindow() { DataContext = new MainWindowViewModel() }
```

**Reply by GoldenWatch** [reply]
Thank you, I found a more complete and working solution example in https://github.com/denisgoriachev/SampleAvaloniaApplication.git, in case there are other newbies like me that needs a working example in Visual Studio to follow and debug thru - it follows the suggestion in the above link by overriding OnFrameworkInitializationCompleted().
